doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,456 @@
+
+1. All reads and writes take place at a provided timestamp.
+2. All reads at timestamp `ts` will return all writes at timestamp ≤ `ts`.
+3. All reads at timestamp `ts` will not return any of the writes at timestamps > `ts`.
+
+As described in the [Timestamp Selection](#Timestamp Selection) section, whenever a read happens at timestamp `ts`, all
+subsequent writes will happen at a timestamp > `ts`. Since a read transaction performs all of it’s read at the same
+timestamp, we can guarantee that subsequent reads will always read the same value since all writes will happen at a
+later timestamp.
+
+#### Serializable
+
+See [https://jepsen.io/consistency/models/serializable](https://jepsen.io/consistency/models/serializable) for an in
+depth formal discussion of Serializable. Serializability prevents Phantom Reads, which means that once a transaction
+reads some value that satisfy some predicate, it will always read the same values, regardless if some other transaction
+has made a write that affects the predicate. Informally this means that you can apply some total order to all
+transactions.
+
+This property holds from how Materialize satisfies [Repeatable Read](#Repeatable Read). All reads in a read only
+transaction happen at the same timestamp. All writes that happen after the first read of a read-only transaction will
+happen at a later timestamp, and therefore will not affect any reads of the read-only transaction.
+
+Informally we can use the transaction timestamps to provide a total ordering over all transactions.
+
+- Transaction T1 is ordered before Transaction T2 if the timestamp of T1 < the timestamp of T2.
+- Transaction T1 is ordered before Transaction T2 if the timestamp of T1 == the timestamp of T2, T1 is write-only, and
+  T2 is read-only.
+- Transaction T1 can be arbitrarily ordered with Transaction T2 if the timestamp of T1 == the timestamp of T2, T1 is
+  read-only, and T2 is read-only.
+
+#### Strict Serializable
+
+See [https://jepsen.io/consistency/models/strict-serializable](https://jepsen.io/consistency/models/strict-serializable)
+for an in depth formal discussion of Strict Serializable. Strict Serializable is a combination of Serializable and
+Linearizable, which means that the total order formed by all transactions must be constrained by their real time
+occurrences. So if transaction t1 happened in real time before transaction t2, then t1 must be ordered before t2 in the
+total ordering.
+
+Operations that only include user tables, and include at least one user table, are currently Strict Serializable. This
+is accomplished by using a single global timestamp and a single thread in the coordinator to order operations. Writes
+happen at a timestamp larger than all previous operations. Reads happen at a timestamp equal to or larger than all
+previous operations.
+
+Operations that involve reading from views over upstream data are not currently Strict Serializable. It is possible to
```

Comment:
"views" isn't quite right, I think "indexes" might be more appropriate? e.g. you can have a materialized source

Reply:
How about "objects"? That may keep it high level.